### ISSUE TITLE
Mish operation PCC error with Float32

### ISSUE THREAD

**Issue by jserbedzijaTT** [tenstorrent/tt-metal#31112]
### Describe the bug

I get PCC error when running the following tests:
```
def test_mish_golden_verification_fixed_tensor(device):
    torch.manual_seed(0)
    
    input_data = torch.tensor([[-1.1258, -1.1524, -0.2506, 1.5863,  0.9463, -0.8437],
                              [-0.6136,  0.0316, -0.4927, -1.2341,  1.8197, -0.5515],
                              [-0.5692,  0.9200,  1.1108, -0.9565,  0.0335,  0.7101]], dtype=torch.float32)
    
    golden_function = ttnn.get_golden_function(ttnn.mish)
    golden_output = golden_function(input_data)
    
    input_tensor = ttnn.from_torch(
        input_data,
        dtype=ttnn.float32,
        layout=ttnn.TILE_LAYOUT,
        device=device
    )
    
    output_tensor = ttnn.mish(input_tensor)
    output_tensor = ttnn.to_torch(output_tensor)
    
    assert_with_pcc(golden_output, output_tensor, pcc=0.99)

```

### Steps to reproduce the issue

Branch: `main`

**Comment by KalaivaniMCW** [comment]
To check with ttnn.hardmish

**Comment by KalaivaniMCW** [comment]
Update: Tested with hardmish , the results are not very accurate for fp32

I made some changes to `log` and tried the formula-approach, the results are better. 
I'll also check with exp61f for fp32
```
golden_output
 tensor([[-0.3082, -0.3086, -0.1302,  1.4973,  0.8090, -0.2896],
        [-0.2501,  0.0193, -0.2186, -0.3086,  1.7503, -0.2349],
        [-0.2394,  0.7818,  0.9824, -0.3005,  0.0205,  0.5710]])

output_tensor with x * tanh(log(1 + exp(x))) with improved log
 tensor([[-0.3094, -0.3098, -0.1302,  1.4969,  0.8086, -0.2904],
        [-0.2492,  0.0193, -0.2178, -0.3099,  1.7503, -0.2340],
        [-0.2385,  0.7813,  0.9825, -0.3011,  0.0205,  0.5708]])
```

<img width="887" height="885" alt="Image" src="https://github.com/user-attachments/assets/011fcb73-2ff8-46bb-86ea-9d9afcda33e9" />

**Comment by KalaivaniMCW** [comment]
Update: 
I tested the new mish op impl  for a specific range (-1e9, 1e9) 
Will update on the perf from soon

<img width="775" height="645" alt="Image" src="https://github.com/user-attachments/assets/adb8bda3-7463-43b6-b34c-f98d55d3a99f" />

**Comment by KalaivaniMCW** [comment]
Update: 
Enabled float32 support for Mish , provided unit test passes.
next steps: to look at improving accuracy with exp21f,  and improved log and tanh

**Comment by KalaivaniMCW** [comment]
Float32 support is available, hence lowering priority.
We'll revisit this once sharding support for eltwise ops is completed
